model.py

```diff
@@ -169,35 +187,25 @@ class Category(db.Model):
 
     __tablename__ = "categories"
 
-    category_id = db.Column(db.Integer, autoincrement=True,
-                            primary_key=True, nullable=False)
-    category_name = db.Column(db.String(100), nullable=False)
+    id = db.Column(db.Integer, primary_key=True, nullable=False)
+    name = db.Column(db.String(100), nullable=False)
+    tasks = db.relationship('Task', secondary=tasks_categories, backref=db.backref('category.id'))
 
     def __repr__(self):
         """Provide helpful representation when printed."""
 
-        return f"<Category id={self.category_id} name={self.category_name}>"
-
+        return f"<Category {self.id} - {self.name}>"
 
-class TaskCategory(db.Model):
-    """Place to store tasks and what categories they belong to."""
-    # Just an association table. Not a meaningful middle table.
-    # A task belongs to a category. It can belong to multiple categories.
-    # A category can have many tasks.
 
-    __tablename__ = "tasks_categories"
-
-    tasks_categories_id = db.Column(db.Integer, autoincrement=True,
-                                    primary_key=True)
-    task_id = db.Column(db.Integer, db.ForeignKey("tasks.task_id"),
-                        nullable=False)
-    category_id = db.Column(db.Integer,
-                            db.ForeignKey("categories.category_id"))
-
-    def __repr__(self):
-        """Provide helpful information when printed."""
+"""Place to store tasks and what categories they belong to."""
+# Just an association table. Not a meaningful middle table.
+# A task belongs to a category. It can belong to multiple categories.
+# A category can have many tasks.
 
-        return f"<TaskCategory id= {self.tasks_categories_id} task_id={self.task_id} category_id={self.category_id}>"
+tasks_categories = db.Table(
```

Comment:
This is a table you no longer have to manage when you do it this way. SQLAlchemy will take care of this for you.

To add a task to a category, you simply use the `Category` instance like so:

`category.tasks.append(task)`

...anywhere in your views.